jobs/payment-jobs/tasks/unpaid_statement_notify_task.py

```diff
@@ -0,0 +1,146 @@
+# Copyright © 2023 Province of British Columbia
+#
+# Licensed under the Apache License, Version 2.0 (the 'License');
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an 'AS IS' BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Task to notify user for any outstanding statement."""
+from datetime import datetime, timedelta
+
+from flask import current_app
+from pay_api.models import db
+from pay_api.models.invoice import Invoice as InvoiceModel
+from pay_api.models.payment_account import PaymentAccount as PaymentAccountModel
+from pay_api.models.statement import Statement as StatementModel
+from pay_api.models.statement_invoices import StatementInvoices as StatementInvoicesModel
+from pay_api.models.statement_recipients import StatementRecipients as StatementRecipientsModel
+from pay_api.models.statement_settings import StatementSettings as StatementSettingsModel
+from pay_api.services.flags import flags
+from pay_api.services.oauth_service import OAuthService
+from pay_api.utils.enums import AuthHeaderType, ContentType, InvoiceStatus, PaymentMethod, StatementFrequency
+from pay_api.utils.util import get_first_and_last_dates_of_month
+from sentry_sdk import capture_message
+from sqlalchemy import func
+
+from tasks.common.enums import OrgStatus
+from utils.auth import get_token
+from utils.mailer import publish_payment_notification
+
+
+class UnpaidStatementNotifyTask:
+    """Task to notify admin for unpaid statements."""
+
+    @classmethod
+    def notify_unpaid_statements(cls):
+        """Notify for unpaid statements with an amount owing."""
+        eft_enabled = flags.is_on('enable-eft-payment-method', default=False)
+
+        if eft_enabled:
+            cls._notify_for_monthly()
+
+    @classmethod
+    def _get_account(cls, auth_token: str, auth_account_id: str):
+        """Return account contact by calling auth API."""
+        get_account_endpoint = current_app.config.get('AUTH_API_ENDPOINT') + f'orgs/{auth_account_id}'
+        account_response = OAuthService.get(get_account_endpoint, auth_token, AuthHeaderType.BEARER, ContentType.JSON)
+        return account_response.json()
+
+    @classmethod
+    def _notify_for_monthly(cls):  # pylint: disable=too-many-locals
+        """Notify for unpaid monthly statements with an amount owing."""
+        # Get the latest monthly statement
+        now = datetime.now()
```

Comment:
Note: datetime.now() might be different locally versus the UTC version on the container

Reply:
Updated to use existing current_local_time() helper which uses timezone config